plugins/outputs/sql/sql.go

```diff
@@ -248,15 +251,36 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 		}
 
 		sql := p.generateInsert(tablename, columns)
-		_, err := p.db.Exec(sql, values...)
 
-		if err != nil {
-			// check if insert error was caused by column mismatch
-			p.Log.Errorf("Error during insert: %v, %v", err, sql)
-			return err
+		switch p.Driver {
+		case "clickhouse":
+			// ClickHouse needs to batch inserts with prepared statements
+			tx, err := p.db.Begin()
+			if err != nil {
+				return fmt.Errorf(": %v", err)
+			}
+			stmt, err := tx.Prepare(sql)
+			if err != nil {
+				return fmt.Errorf("Error during prepare: %v", err)
+			}
+			defer stmt.Close() //nolint
```

Comment:
Please limit the nolint scope. I guess
```suggestion
			//nolint:revive // We cannot do anything about a failing close.
			defer stmt.Close()
```
will do.